modin/pandas/test/test_general.py

```diff
@@ -105,50 +114,66 @@ def test_merge():
     modin_df2 = pd.DataFrame(frame_data2)
     pandas_df2 = pandas.DataFrame(frame_data2)
 
-    join_types = ["outer", "inner"]
-    for how in join_types:
-        # Defaults
-        modin_result = pd.merge(modin_df, modin_df2, how=how)
+    join_type_outer = "outer"
+    join_type_inner = "inner"
+    for how in (join_type_outer, join_type_inner):
+        if how == join_type_outer:
+            warning_catcher = warns_that_defaulting_to_pandas()
+        else:
+            warning_catcher = contextlib.nullcontext()
+        with warning_catcher:
+            modin_result = pd.merge(modin_df, modin_df2, how=how)
```

Comment:
why got rid of the idea of storing join types in a list?
```suggestion
    join_types = ["outer", "inner"]
    for how in join_types :
        with warns_that_defaulting_to_pandas() if how == "outer" else contextlib.nullcontext():
            modin_result = pd.merge(modin_df, modin_df2, how=how)
```